### ISSUE TITLE
weather widget does not respect "precision" setting

### ISSUE THREAD

**Issue by snjoetw** [home-assistant/appdaemon#321]
I had precision set to "0" in the global_parameters and I see climate does respect my setting but not weather.

I wonder if this is as simple as updating this line 
https://github.com/home-assistant/appdaemon/blob/a476b021a9745c3555897f41586244fb8c1fed18/appdaemon/widgets/baseweather/baseweather.js#L88

to

```
self.set_field(self, field, self.format_number(self, state.state))
```

I'm not sure how to test appdaemon locally so I can't tell if that'd cause side effect or not, e.g. non-number fields will mess up (but is there non-number field in weather widget?)

**Comment by ReneTode** [comment]
i did take a look and i think it can work, because as i see it there are just number fields in the widget.
but i must leave it up to @acockburn to test if i am right, because i dont heave weather widgets.

**Comment by acockburn** [comment]
Indeed it is that simple :) This will be fixed in AD 3.0.3
